models/cow_protocol/cow_protocol_trades.sql

```diff
@@ -0,0 +1,42 @@
+{{ config(
+        alias ='trades',
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "cow_protocol",
+                                    \'["bh2smith", "gentrexha"]\') }}'
+        )
+}}
+SELECT *
+FROM
+(
+        SELECT
+            'ethereum' AS blockchain,
+            'CoW Protocol' AS project,
+            '1' AS version,
+            TRY_CAST(date_trunc('DAY', block_time) AS date) AS block_date,
```

Comment:
@bh2smith @gentrexha what do you think about moving this block_date logic to the base object ```cow_protocol_ethereum.trades``` and add partition in the config block? this has been an approach we've taken on dex trades, with the idea that the column will be used in downstream queries/dashboards often to filter on date ranges

our base tables (like ethereum transactions) are partitioned in this format, so i thought it'd make sense to match that -- could help query performance

Reply:
Sounds like a good idea. I had noticed this pattern lingering around the project and totally makes sense.

Reply:
is there any good reason for using "try_cast" here?

Reply:
no particular reason, an old habit i guess. we would actually want it to fail if incompatible, but we know the block_time coming through will be fine to cast as date, so no big deal currently